🆔 Add custom reference to erp invoice

daniel-rose opened the pull request:
- prepare transfer definition to add "custom_reference" to "ErpInvoiceTransfer" and "RestErpInvoiceTransfer"
- add column "custom_reference" to database table "foo_erp_invoice"
- expand result search data with "custom_reference"
- ...

julianzimmermann commented on bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Business/Handler/ErpOrderTotalHandler.php:
```diff
@@ -0,0 +1,96 @@
+            $erpOrderTotalTransfer = $erpOrderTransfer->getTotal()
+                ->setFkErpOrder($erpOrderTransfer->getIdErpOrder());
+
+            $erpOrderTotalTransfer = $this->create($erpOrderTotalTransfer);
```
maybe early return instead of else...

julianzimmermann commented on bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderEntityManager.php:
```diff
@@ -71,6 +73,25 @@ public function createErpOrderAddress(ErpOrderAddressTransfer $orderAddressTrans
+            ->requireTaxTotal();
+
+        $entity = new ErpOrderTotal();
+        $entity->fromArray($orderTotalTransfer->toArray());
```
modifiedToArray maybe better?

julianzimmermann commented on bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderEntityManager.php:
```diff
@@ -260,6 +281,36 @@ public function updateErpOrderAddress(ErpOrderAddressTransfer $erpOrderAddressTr
+            ));
+        }
+        $id = $entity->getIdErpOrderTotal();
+        $entity->fromArray($erpOrderTotalTransfer->toArray());
```
modifiedToArray maybe better?

daniel-rose committed:
🆔 Add custom reference to erp invoice
- prepare transfer definition to add "custom_reference" to "ErpInvoiceTransfer" and "RestErpInvoiceTransfer"
- add column "custom_reference" to database table "foo_erp_invoice"
- expand result search data with "custom_reference"
- ...
